modules/web/js/ballerina/diagram2/views/default/components/transform/transform-node-mapper.js

```diff
@@ -876,6 +882,9 @@ class TransformNodeMapper {
                     const simpleVarRefExpression = TransformFactory.createVariableRefExpression(outputVarName);
                     stmt.replaceVariablesByIndex(index, simpleVarRefExpression, true);
                 }
+            } else if (TreeUtil.isFieldBasedAccessExpr(stmt.getVariables()[0])
+                        && stmt.getVariables()[0].getExpression().getVariableName().getValue() === expStr) {
```

Comment:
getSource() ??

Reply:
@maheeka 
when removing a struct, expr will be struct object name. if we use getSource() it will return <structName>.<fieldName>. So instead this is checking VariableName against expr for FieldBasedAccessExpr